### PR TITLE
Add getImplicitRolesForUser and getImplicitPermissionsForUser

### PR DESCRIPTION
**Pull request by nodece**

### CODE REVIEW

File: test/rbacAPI.test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+it('test getImplicitRolesForUser with domain', async () => {
+  const e = await newEnforcer('examples/rbac_with_domains_model.conf', 'examples/rbac_with_hierarchy_with_domains_policy.csv');
+  expect(e.getImplicitRolesForUser('alice', 'domain1')).toEqual(['role:global_admin', 'role:reader', 'role:writer']);
+});
+
+it('test getImplicitPermissionsForUser', async () => {
+  const e = await newEnforcer('examples/rbac_model.conf', 'examples/rbac_with_hierarchy_policy.csv');
+  expect(e.getImplicitPermissionsForUser('bob')).toEqual([['bob', 'data2', 'write']]);
+  expect(e.getImplicitPermissionsForUser('alice')).toEqual([['alice', 'data1', 'read'], ['data1_admin', 'data1', 'read'], ['data1_admin', 'data1', 'write'], ['data2_admin', 'data2', 'read'], ['data2_admin', 'data2', 'write']]);
+});
```
Comment (hsluoyz) [inline test/rbacAPI.test.ts]: Add a trailing empty line please.

File: src/rbac/defaultRoleManager.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ export class DefaultRoleManager implements RoleManager {
     }
 
     if (!this.hasRole(name1) || !this.hasRole(name2)) {
-      throw new Error('error: name1 or name2 does not exist');
+      return;
```
Comment (hsluoyz) [inline src/rbac/defaultRoleManager.ts]: Why remove the error?

File: src/rbac/defaultRoleManager.ts [hunk 3]
```diff
@@ -90,7 +90,7 @@ export class DefaultRoleManager implements RoleManager {
     }
 
     if (!this.hasRole(name)) {
-      throw new Error('error: name does not exist');
+      return [];
```
Comment (hsluoyz) [inline src/rbac/defaultRoleManager.ts]: Why remove the error?

File: src/rbac/defaultRoleManager.ts [hunk 4]
```diff
@@ -113,7 +113,7 @@ export class DefaultRoleManager implements RoleManager {
...
```
Comment (hsluoyz) [inline src/rbac/defaultRoleManager.ts]: Why remove the error?

File: src/rbac/defaultRoleManager.ts (hunk 2 above)
Comment (nodece) [reply src/rbac/defaultRoleManager.ts]: I think it is not an error,  so remove this error.

File: src/rbac/defaultRoleManager.ts (hunk 3 above)
Comment (nodece) [reply src/rbac/defaultRoleManager.ts]: I think it is not an error,  so remove this error.

File: src/rbac/defaultRoleManager.ts (hunk 4 above)
Comment (nodece) [reply src/rbac/defaultRoleManager.ts]: I think it is not an error,  so remove this error.

File: src/rbac/defaultRoleManager.ts (hunk 3 above)
Comment (nodece) [reply src/rbac/defaultRoleManager.ts]: I think it is not an error,  so remove this error.

**Pull request merged**